pandas/core/base.py

```diff
@@ -517,9 +517,16 @@ def _make_str_accessor(self):
             raise AttributeError("Can only use .str accessor with string "
                                  "values, which use np.object_ dtype in "
                                  "pandas")
-        elif isinstance(self, Index) and self.inferred_type != 'string':
-            raise AttributeError("Can only use .str accessor with string "
-                                 "values (i.e. inferred_type is 'string')")
+        elif isinstance(self, Index):
+            # see scc/inferrence.pyx which can contain string values
+            allowed_types = ('string', 'unicode', 'mixed', 'mixed-integer')
+            if self.inferred_type not in allowed_types:
+                message = ("Can only use .str accessor with string values "
+                           "(i.e. inferred_type is 'string', 'unicode' or 'mixed')")
```

Comment:
I don't think you can accept ``mixed-integer``, as these are in general python integer objects which need stringification first (which we *could* do, but that's a separate issue).

```
In [10]: pd.lib.infer_dtype([1,2,'a'])
Out[10]: 'mixed-integer'
```